tests/components/technove/test_number.py

```diff
@@ -0,0 +1,210 @@
+"""Tests for the TechnoVE number platform."""
+
+from unittest.mock import MagicMock
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+from technove import TechnoVEConnectionError, TechnoVEError
+
+from homeassistant.components.number import (
+    ATTR_VALUE,
+    DOMAIN as NUMBER_DOMAIN,
+    SERVICE_SET_VALUE,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_UNAVAILABLE, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
+from homeassistant.helpers import entity_registry as er
+
+from . import setup_with_selected_platforms
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default", "mock_technove")
+async def test_numbers(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test the creation and values of the TechnoVE numbers."""
+    await setup_with_selected_platforms(hass, mock_config_entry, [Platform.NUMBER])
+
+    entity_entries = er.async_entries_for_config_entry(
+        entity_registry, mock_config_entry.entry_id
+    )
+
+    assert entity_entries
+    for entity_entry in entity_entries:
+        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
+        assert (state := hass.states.get(entity_entry.entity_id))
+        assert state == snapshot(name=f"{entity_entry.entity_id}-state")
```

Comment:
Why don't you just use snapshot() and let the framework handle the data?

Reply:
I'm sorry I don't understand what you mean, do you have an example I could lookup?

This is how I've done snapshot test for my other entity types (switch, sensor, etc) and that's how I've been told during my PRs. Is there a better way?

I did a quick search in the code and I found this recent example, this seems simpler to do, is that an example of best practice?
https://github.com/joostlek/core/blob/ef1c91c4cf03afa044547d627781a37b461ab5e8/tests/components/airgradient/test_switch.py#L31-L42

Reply:
I implemented the changes based on the example I found for airgradient, it is much simpler. Let me know if this is not what you had in mind. Thanks!